Raven.NET.Core.Tests/ProviderTests/RavenProviderTests.cs

```diff
@@ -0,0 +1,213 @@
+        sut.AddRaven(name, raven);
+        
+        //Assert
+        RavenCache.RavenWatcherCache.ShouldContainKey(name);
+        
+        RavenCache.RavenWatcherCache.Clear();
+    }
+    
+    [Fact]
+    void AddRaven_Should_ThrowException_WhenRavenAlreadyExists()
+    {
+        //Arrange 
+        var raven = _ravenWatcher.Object;
+        var name = "TestWatcher";
+        RavenCache.RavenWatcherCache.TryAdd(name, raven);
+
+        //Act
+        var ex = Record.Exception(() => sut.AddRaven(name, raven));
+        
+        //Assert
+        ex.ShouldNotBeNull();
+        ex.ShouldBeOfType<RavenAlreadyExistsException>();
+        
+        RavenCache.RavenWatcherCache.Clear();
+
+    }
+    
+    [Fact]
+    void AddRaven_Should_AddRavenToInternalRavenTypeCache_WhenTypeProvided()
+    {
+        //Arrange 
+        RavenCache.RavenTypeWatcherCache.Clear();
+        var raven = _ravenTypeWatcher.Object;
+        var name = "TestWatcher";
+
+        //Act
+        sut.AddRaven(name, raven, typeof(TestSubjectEntity));
+        
+        //Assert
+        RavenCache.RavenTypeWatcherCache.ShouldContainKey(typeof(TestSubjectEntity));
+    }
+    
+    [Fact]
+    void AddRaven_Should_ThrowException_WhenRavenTypeAlreadyExists()
+    {
+        //Arrange 
+        var raven = _ravenTypeWatcher.Object;
+        var name = "TestWatcher";
+        RavenCache.RavenTypeWatcherCache.TryAdd(typeof(TestSubjectEntity), raven);
+
+        //Act
+        var ex = Record.Exception(() => sut.AddRaven(name, raven, typeof(TestSubjectEntity)));
+        
+        //Assert
+        ex.ShouldNotBeNull();
+        ex.ShouldBeOfType<RavenForTypeAlreadyExistsException>();
+        
+        RavenCache.RavenTypeWatcherCache.Clear();
+    }
+    
+    [Fact]
+    void RemoveRaven_Should_RemoveRavenFromInternalCache()
+    {
+        //Arrange 
+        var raven = _ravenWatcher.Object;
+        var name = "TestWatcher";
+        RavenCache.RavenWatcherCache.TryAdd(name, raven);
+
+        //Act
+        sut.RemoveRaven(name);
+        
+        //Assert
+        RavenCache.RavenWatcherCache.ShouldBeEmpty();
+    }
+    
+    [Fact]
+    void RemoveRaven_Should_ThrowExceptionWhenRavenDoesNotExist()
+    {
+        //Arrange 
+        RavenCache.RavenWatcherCache.Clear();
+        var raven = _ravenWatcher.Object;
+        var name = "TestWatcher";
+
+        //Act
+        var ex = Record.Exception(() => sut.RemoveRaven(name));
+        
+        //Assert
+        ex.ShouldNotBeNull();
+        ex.ShouldBeOfType<RavenDoesNotExistsException>();
+    }
+    
+    [Fact]
+    void GetRaven_Should_ReturnRavenWatcher_WhenNoTypeProvided()
+    {
+        //Arrange 
+        var raven = _ravenWatcher.Object;
+        var name = "TestWatcher";
+        RavenCache.RavenWatcherCache.TryAdd(name, raven);
+        
+        //Act
+        var result = sut.GetRaven(name);
+        
+        //Assert
+        result.ShouldNotBeNull();
+        result.ShouldBeEquivalentTo(raven);
```

Comment:
ShouldBe is not enough? just a question :)

Reply:
Correct, I changed it ;)